README.md

```diff
@@ -2,33 +2,651 @@
+    },
+};
+
+const headers = {
+    'custom-header': 'header-value',
+};
+
+const result = fetchFromApi('http://domain:8000/graphql/api', body, contentPath, headers,);
+```
+
+Response type:
+
+```tsx
+type GuillotineResult = {
+    error?: {
+        code: string,
+        message: string
+    } | null;
+    [dataKey: string]: any;
+};
+```
+
+---
+
+### `fetchFromApi(apiUrl: string, body: ContentApiBaseBody, headers?: {}, method?: string) => Promise<any>`
+
+Makes custom third-party service request. Used by the [fetchGuillotine()](#fetch-guillotine) method.
+
+| Argument              | Description                                        |
+|-----------------------|----------------------------------------------------|
+| `apiUrl`              | Service API URL                                    |
+| `body`                | Body of the request                                |
+| `headers = undefined` | Object of custom headers with string values        |
+| `method = 'POST'`     | Method of the request: `HEAD`, `POST`, `GET`, etc. |
+
+Usage:
+
+```tsx
+import {fetchFromApi} from '@enonic/nextjs-adapter';
+
+const body = {
+    query: 'qraphql query as string',
+    variables: {
+        foo: 'bar',
+    },
+}
+
+const headers = {
+    'custom-header': 'header-value',
+}
+
+const result = fetchFromApi('http://domain:8000/graphql/api', body, headers, 'GET');
+```
+
+---
+
+### `getUrl(url: string, meta: MetaData) => string`
+
+Converts an absolute URL to the relative one for current viewer (Next.js/Enonic XP).
+
+> **INFO:** For your URLs to work both in Enonic XP and Next.js you need to:
+> 1. Query absolute URLs from guillotine
+> 2. Wrap them with `getUrl()` function in the views
+
+| Argument | Description                                             |
+|----------|---------------------------------------------------------|
+| `url`    | URL you want to transform                               |
+| `meta`   | Runtime data returned by [fetchContent](#fetch-content) |
+
+Usage:
+
+```tsx
+import {getUrl} from '@enonic/nextjs-adapter';
+
+const urlRelativeToViewer = getUrl('http://domain:8000/graphql/api', meta);
+```
+
+---
+
+### <a id="rich-text-query"></a>`richTextQuery(fieldName: string) => string`
+
+This is a utility function for querying for `RichTextData` needed for [RichTextView](#rich-text-view). It creates a graphql query string for
+HTML area input type with given field name.
+
+| Argument    | Description          |
+|-------------|----------------------|
+| `fieldName` | HTML area field name |
+
+Usage:
+
+```tsx
+import {richTextQuery} from '@enonic/nextjs-adapter';
 
+const query = `query($path:ID!){
+    guillotine {
+        get(key:$path) {
+            _path
+            type
+            ${richTextQuery('htmlField')}
+        }
+    }
+}`;
 ```
 
-## Problems
+### `ComponentRegistry`
+
+Registry containing definitions of all components (i.e. pages, parts, layouts, macros, etc. ). It is used in the runtime by `nextjs-adapter`
```

Comment:
Unnecessary linebreak?

Reply:
Doesn't affect the output, but easier to navigate the source